shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.m

```diff
@@ -23,4 +29,172 @@ - (void)testBinaryMessenger {
   OCMVerify([engine binaryMessenger]);
 }
 
+- (void)testItReportsLightPlatformBrightnessByDefault {
+  // Setup test.
+  id engine = OCMClassMock([FlutterEngine class]);
+
+  id settingsChannel = OCMClassMock([FlutterBasicMessageChannel class]);
+  OCMStub([engine settingsChannel]).andReturn(settingsChannel);
+
+  FlutterViewController* vc = [[FlutterViewController alloc] initWithEngine:engine
+                                                                    nibName:nil
+                                                                     bundle:nil];
+
+  // Exercise behavior under test.
+  [vc traitCollectionDidChange:nil];
+
+  // Verify behavior.
+  OCMVerify([settingsChannel sendMessage:[OCMArg checkWithBlock:^BOOL(id message) {
+                               return [message[@"platformBrightness"] isEqualToString:@"light"];
+                             }]]);
+}
+
+- (void)testItReportsPlatformBrightnessWhenViewWillAppear {
+  // Setup test.
+  id engine = OCMClassMock([FlutterEngine class]);
```

Comment:
Have you verified that `engine` won't outlive this test? Per offline discussion, if a class mock outlives the test, Bad Things happen. I think the recent refactors to method channel handling means that it won't leak the engine like it used to, but I'd like to be sure.

Reply:
How might I verify this?

Reply:
I just went ahead and explicitly cleaned up every mock in each test.